Objects/unicodeobject.c

```diff
@@ -6011,9 +6011,18 @@ _PyUnicode_DecodeUnicodeEscapeInternal(const char *s,
                 if (namelen) {
                     /* found a name.  look it up in the unicode database */
                     s++;
+
+                    /* map to uppercase before calling getcode, which does
+                     * case-sensitive comparisons */
+                    char buffer[namelen];
+                    size_t i;
+                    for (i = 0; i < namelen; i++) {
+                        buffer[i] = Py_TOUPPER(start[i]);
+                    }
```

Comment:
This is similar transformation to [one inside of `unicodedata`](https://github.com/python/cpython/blob/33838fedf7ed6ee907a49d042f8fa123178a1f9c/Modules/unicodedata.c#L1167-L1170), but does not quite replace it. I tried removing that clause and saw a failure in `test_re`.